packages/integration_test/example/integration_test/_example_test_web.dart

```diff
@@ -21,15 +23,74 @@ void main() {
     // Trigger a frame.
     await tester.pumpAndSettle();
 
+    final Finder finder = find.byKey(const Key('platform'));
+    final Text platformText = tester.widget(finder);
+
     // Verify that platform is retrieved.
-    expect(
-      find.byWidgetPredicate(
-        (Widget widget) =>
-            widget is Text &&
-            widget.data
-                .startsWith('Platform: ${html.window.navigator.platform}\n'),
-      ),
-      findsOneWidget,
-    );
+    expect(finder, findsOneWidget);
+    expect(platformText.data, 'Platform: ${html.window.navigator.platform}\n');
+  });
+
+  testWidgets('verify event dispatching works', (WidgetTester tester) async {
+    // Build our app and trigger a frame.
+    app.main();
+
+    // Trigger a frame.
+    await tester.pumpAndSettle();
+
+    final Finder finderButton = find.byKey(const Key('button'));
+    expect(finderButton, findsOneWidget);
+
+    final Offset offset = tester.getCenter(finderButton);
+
+    dispatchEventToMouseLocation((offset.dx).toInt(), (offset.dy).toInt());
+    await tester.pumpAndSettle();
+
+    final Finder finder = find.byKey(const Key('counter'));
+    final Text counter = tester.widget(finder);
+
+    // Verify that counter is clicked.
+    expect(finder, findsOneWidget);
+    expect(counter.data, '1');
```

Comment:
Actually, can we assert earlier that this is '0'?

Reply:
Good idea! I'll make the change thanks :)